addons/im_livechat/static/src/new/core/thread_service.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module */
+
+import { ThreadService } from "@mail/new/core/thread_service";
+import { removeFromArray } from "@mail/new/utils/arrays";
+import { assignDefined, createLocalId } from "@mail/new/utils/misc";
+import { patch } from "@web/core/utils/patch";
+
+patch(ThreadService.prototype, "im_livechat", {
+    insert(data) {
+        const isUnknown = !(createLocalId(data.model, data.id) in this.store.threads);
+        const thread = this._super(data);
+        if (thread.type === "livechat") {
+            if (data.serverData) {
+                assignDefined(thread, data.serverData?.channel, ["anonymous_name"]);
+            }
+            if (isUnknown) {
+                this.store.discuss.livechat.threads.push(thread.localId);
+            }
```

Comment:
It's annoying to have this kind of process, shouldn't livechat threads be a compute/getter instead?

Reply:
I agree, but this could be done in a follow up PR for every type of threads if it's fine for you (chats/channels/livechats).